### ISSUE TITLE
Certain hot-keys not accepted by Preferences UI

### ISSUE THREAD

**Issue by Wedge009** [wesnoth/wesnoth#3000]
The only custom hot-key I use regularly is for End Turn. I prefer Shift+Enter to End Turn instead of the default Ctrl+Space because I think I've mixed it up with Space (to end unit's moves) before. (I know there's an undo... whatever the reason, I should be able to specify a different hot-key.)

With the version change from 1.12/1.13 to 1.14/1.15, preferences got moved around, etc. I tried to recreate my custom hot-key but I'm not able to any more - I know this was working fine in 1.13.

Updating the preference file manually (with Wesnoth not running) seems to be okay. Tested in 1.14.0 on Windows and Linux.

**Comment by Wedge009** [comment]
Heh, 3000th issue...

**Comment by jostephd** [comment]
Reproduced in current master.

**Comment by Wedge009** [comment]
I'm not sure if things have worsened since I last looked at this, but even testing single letters seems to be inconsistent. For example, if I try to add `o` as a hot-key assignment it correctly warns me that it's currently in use. If I try to add `y`, however, it gives me no warning and happily adds the assignment even though it's already assigned to 'Execute Action'.

1.14.9 release.

**Comment by jostephd** [comment]
In master:

![2019-10-21-235659_328x136_scrot](https://user-images.githubusercontent.com/24784687/67251382-7b0c0800-f45e-11e9-8af7-280558ab0a85.png)

Here I tried to add `o` as the hotkey to `Next tool`. Note that both `Next tool` and `Draw hex coordinators` are bindings for editor mode. If you had tried to bind `o` to a game-mode command (mode `G` rather than `E`), no warnings would be correct behavior.

**Comment by Wedge009** [comment]
Yes, I did consider the different domains, but maybe I missed the destination domain. Hopefully that should be okay then.

The issue about special keys like Enter is still outstanding, though.

**Comment by Wedge009** [comment]
Not a big deal for me, but noting it's still a problem for 1.15.17+dev (1.16 beta).

**Comment by Wedge009** [comment]
I suppose I got used to the default Ctrl+Space at some point, but I just retested 1.18 and 1.19 and Shift+Enter is accepted again.
